### DISCUSSION TITLE
Customize the StandardListView widget

### DISCUSSION THREAD

**Discussion by Horbin-Magician** [slint-ui/slint#3299] [Q&A] [answered]
StandardListView is a very useful widget. But it's not customizable enough, i.e. I can't show an icon.

I want to know how to deeply customize this widget, or where can I see the source code for this component?

**Comment by hunger** [comment, marked as answer]
The `StandardListViewItem` in `internal\common\builtin_structs.rs` defines just a `text`. You probably need to extend that to include more information there so that the widgets can make use of it.

The widgets themselves are defined in the widgets styles in `internal/compiler/widgets/...`.

**Reply by Horbin-Magician** [reply]
Many thanks!
